scripts/dc2/run_dc2_sprinkler.py

```diff
@@ -0,0 +1,391 @@
+                                                             glsne_ddf_final,
+                                                             os.path.join(output_dir, 'lensed_sne_truth.db'),
+                                                             return_df=True, overwrite_existing=True)
+
+
+if __name__ == '__main__':
+
+    catalog_version = 'cosmoDC2_v1.1.4_image_addon_knots'
+    agn_db = '/global/cfs/cdirs/lsst/projecta/lsst/groups/SSim/DC2/cosmoDC2_v1.1.4/agn_cosmoDC2_v1.1.4.db'
+    sed_dir = '/global/projecta/projectdirs/lsst/groups/SSim/DC2/cosmoDC2_v1.1.4/sedLookup'
+
+    parser = argparse.ArgumentParser(
+        description='Run DC2 Sprinkler and generate truth catalogs.'
+    )
+    parser.add_argument('--input_dir', type=str, help='Input Data Directory')
+    parser.add_argument('--checkpoint_dir', type=str, help='Checkpoint File Directory')
+    parser.add_argument('--output_dir', type=str, help='Output Data Directory')
+    args = parser.parse_args()
+
+    # Load DC2 catalogs
+    if os.path.exists(os.path.join(args.checkpoint_dir, 'dc2_lens_ckpt.csv')):
+        dc2_lenses = pd.read_csv(os.path.join(args.checkpoint_dir, 'dc2_lens_ckpt.csv'))
+    else:
+        dc2_lenses = load_dc2_lenses(catalog_version, agn_db)
+        dc2_lenses.to_csv(os.path.join(args.checkpoint_dir, 'dc2_lens_ckpt.csv'), index=False)
+
+    if (os.path.exists(os.path.join(args.checkpoint_dir, 'dc2_agn_host_ckpt.csv')) &
+        os.path.exists(os.path.join(args.checkpoint_dir, 'dc2_sne_host_ckpt.csv'))):
+        dc2_agn_hosts = pd.read_csv(os.path.join(args.checkpoint_dir, 'dc2_agn_host_ckpt.csv'))
+        dc2_sne_hosts = pd.read_csv(os.path.join(args.checkpoint_dir, 'dc2_sne_host_ckpt.csv'))
+    else:
+        dc2_agn_hosts, dc2_sne_hosts = load_dc2_hosts(catalog_version, agn_db, sed_dir)
+        dc2_agn_hosts.to_csv(os.path.join(args.checkpoint_dir, 'dc2_agn_host_ckpt.csv'), index=False)
+        dc2_sne_hosts.to_csv(os.path.join(args.checkpoint_dir, 'dc2_sne_host_ckpt.csv'), index=False)
+
+    # Load AGN Lens Cat
+    om10_hdu = fits.open(os.path.join(args.input_dir, 'om10_qso_mock.fits'))
+    om10_data = om10_hdu[1].data
+    # Only keep om10 systems where the host galaxy is within the redshift range of cosmoDC2
+    om10_data = om10_data[np.where(om10_data['ZSRC'] <= 3.0)]
+
+    # Load SNe Lens Cat
+    glsne_df_system = pd.read_hdf(os.path.join(args.input_dir, 'glsne_dc2_v2.h5'), key='system')
+    glsne_df_image = pd.read_hdf(os.path.join(args.input_dir, 'glsne_dc2_v2.h5'), key='image')
+    glsne_merged_df = merge_catalog(glsne_df_system, glsne_df_image)
+    # Remove objects where the host galaxy is more than 1.25 arcseconds
+    # from the source so we can image it with postage stamp code
+    glsne_query = 'host_x < 1.25 and host_y < 1.25 and host_x > -1.25 and host_y > -1.25'
+    glsne_merged_df = glsne_merged_df.query(glsne_query).reset_index(drop=True)
+
```

Comment:
We need to add a line
```
    os.makedirs(args.output_dir, exist_ok=True)`
```
before calling `run_dc2_sprinkler`, otherwise the code will crash when it tries to write the output files.